advance_check_void/models/account_payment.py

```diff
@@ -0,0 +1,391 @@
+                        rec.check_number,
+                        rec.state.title(),
+                    )
+                    rec.message_post(body=message)
+        result = super(AccountPayment, self).print_checks()
+        return result
+
+    def void_check_button(self):
+        return {
+            "name": _("Void Check"),
+            "type": "ir.actions.act_window",
+            "res_model": "simple.void.check",
+            "view_type": "form",
+            "view_mode": "form",
+            "target": "new",
+            "context": {
+                "default_payment_id": self.id,
+            },
+        }
+
+    def write(self, vals):
+        if vals.get("check_number") and not str(vals.get("check_number")).isdigit():
+            raise ValidationError(_("Check number must be integer."))
+        result = super(AccountPayment, self).write(vals)
+        check_hist_obj = self.env["payment.check.history"]
+        for res in self:
+            new_chk_id = False
+            self._cr.execute(
+                "update account_payment set write_uid=%s where id=%s",
+                (
+                    self._uid,
+                    res.id,
+                ),
+            )
+            if "check_number" in vals and vals["check_number"] and res.check_number:
+                check_hist = {
+                    "name": res.name,
+                    "payment_id": res.id,
+                    "partner_id": res.partner_id.id,
+                    "amount": res.amount,
+                    "check_amount": res.amount,
+                    "check_number": res.check_number,
+                    "journal_id": res.journal_id.id,
+                    "date": fields.Date.context_today(res),
+                    "create_date": fields.Date.context_today(res),
+                    "write_date": fields.Date.context_today(res),
+                    "create_uid": res.create_uid.id,
+                    "state": res.state,
+                    "is_visible_check": True if not res.check_number else False,
+                }
+                new_chk_id = check_hist_obj.create(check_hist)
+            if new_chk_id:
+                check_ids = check_hist_obj.search(
+                    [
+                        ("payment_id", "=", res.id),
+                        ("check_number", "=", res.check_number),
+                        ("id", "!=", new_chk_id.id),
+                    ],
+                    order="id DESC",
+                    limit=1,
+                )
+            else:
+                check_ids = check_hist_obj.search(
+                    [("payment_id", "=", res.id), ("write_date", "<=", res.write_date)],
+                    order="id desc",
+                    limit=1,
+                )
+            if check_ids:
+                for chk in check_ids:
+                    if res.state == "sent":
+                        continue
+                    chk.write({"state": "void"})
```

Comment:
Why the additonal line?
```suggestion
                    if res.state != "sent":
                        chk.write({"state": "void"})
```